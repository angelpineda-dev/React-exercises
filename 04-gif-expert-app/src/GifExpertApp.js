import React, { useState } from "react";
import { AddCategory } from "./Components/AddCategory";
import { GifGrid } from "./Components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  // Default value of search.
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
