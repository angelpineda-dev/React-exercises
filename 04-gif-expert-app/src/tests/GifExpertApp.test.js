import React from "react";
import { GifExpertApp } from "../GifExpertApp";
import { shallow } from "enzyme";

describe("Pruebas en  <GifExpertApp/>", () => {
  let wrapper = shallow(<GifExpertApp />);

  test("Debe mostrar el componente correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de categorias", () => {
    let categories = ["Goku", "Seiya"];
    let wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
