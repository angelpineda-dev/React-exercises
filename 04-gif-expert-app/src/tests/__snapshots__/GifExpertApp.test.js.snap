// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Seiya"
      key="Seiya"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en  <GifExpertApp/> Debe mostrar el componente correctamente. 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
